src/layers/components/BasketModal/BasketModal.ts

```diff
@@ -0,0 +1,357 @@
+            </div>
+            <div class="BasketModal__form-item">
+              <div class="BasketModal__form-header">Address</div>
+              <input class="BasketModal__input address" type='text'/>
+              <div class="BasketModal__error address"></div>
+            </div>
+            <div class="BasketModal__form-item">
+              <div class="BasketModal__form-header">Email</div>
+              <input class="BasketModal__input email" type='text'/>
+              <div class="BasketModal__error email"></div>
+            </div>
+          </div>
+          <div class="BasketModal__card-block">
+            <h2 class="BasketModal__card-header">Credit card details</h2>
+            <div class="BasketModal__card">
+              <div class="BasketModal__firstLine">
+                <img class="BasketModal__cardImg" src="https://i.guim.co.uk/img/media/b73cc57cb1d46ae742efd06b6c58805e8600d482/16_0_2443_1466/master/2443.jpg?width=700&quality=85&auto=format&fit=max&s=fb1dca6cdd4589cd9ef2fc941935de71"/>
+                <input class="BasketModal__cardNumber"/>
+              </div>
+              <div class="BasketModal__secondLine">
+                <div class="BasketModal__valid">
+                  VALID: 
+                  <input class="BasketModal__valid-input"/>
+                </div>
+                <div class="BasketModal__cvv">
+                  CVV: 
+                  <input class="BasketModal__cvv-input"/>
+                </div>
+              </div>
+            </div>
+            <span class="BasketModal__cardNumber-error BasketModal__error"></span>
+            <span class="BasketModal__cardValid-error BasketModal__error"></span>
+            <span class="BasketModal__cardCVV-error BasketModal__error"></span>
+            <button class="BasketModal__button button button-basic">Confirm</button>
+          </div>
+        `;
+
+        const basketModal = document.createElement('div');
+        basketModal.classList.add('BasketModal');
+
+        basketModal.innerHTML = modal;
+        this.container.append(basketModal);
+
+        const buttonConfirm = document.querySelector('.BasketModal__button');
+        if (buttonConfirm) {
+            buttonConfirm.addEventListener('click', () => this.confirm());
+        }
+
+        const cardNumberInput: HTMLInputElement | null = document.querySelector('.BasketModal__cardNumber');
+        const cardNumberImage: HTMLImageElement | null = document.querySelector('.BasketModal__cardImg');
+        if (cardNumberInput) {
+            cardNumberInput.addEventListener('input', (e: Event) => {
+                if (e.target instanceof HTMLInputElement) {
+                    const value = e.target.value;
+                    const arr = value.split(' ');
+
+                    if (Number.isNaN(Number(arr[arr.length - 1]))) {
+                        cardNumberInput.value = `${value.slice(0, -1)}`;
+                    }
+                    if (cardNumberImage) {
```

Comment:
I advise you to split large functions into several atomic ones. For example, the contents of this if block could be placed separately in the renderCardNumberImage function

Reply:
Update